miso-web/src/main/webapp/scripts/warning_pool.js

```diff
@@ -22,61 +22,52 @@
-        return dilution.identityConsentLevel === 'Revoked';
-      }), "(CONSENT REVOKED)"]
-      ], warnings);
-    return Warning.generateTableWarnings(data, warnings);
-  },
-  tileWarnings: function(pool){
-    var warnings = [];
-    warnings = Warning.addWarnings([
-      [pool.prioritySubprojectAliases && pool.prioritySubprojectAliases.length > 0,
-        'PRIORITY ('
-        + (pool.prioritySubprojectAliases.length == 1 ? pool.prioritySubprojectAliases[0] :
-          pool.prioritySubprojectAliases.join(', ')) + ')', "info"],
-      [pool.duplicateIndices, "DUPLICATE INDICES"],
-      [pool.nearDuplicateIndices && !pool.duplicateIndices, "NEAR-DUPLICATE INDICES"],
-      [pool.hasEmptySequence, "MISSING INDEX"],
-      [pool.hasLowQualityLibraries, "LOW QUALITY LIBRARIES"],
-      [pool.pooledElements && pool.pooledElements.some(function(dilution){
-        return dilution.identityConsentLevel === 'Revoked';
-      }), "CONSENT REVOKED"]
-      ], warnings);
-    return Warning.generateTileWarnings(warnings);
-  },
-  
+    return [
+        {
+          include: pool.prioritySubprojectAliases && pool.prioritySubprojectAliases.length > 0,
+          headerMessage: 'Belongs to high priority subproject'
+              + (pool.prioritySubprojectAliases.length == 1 ? ' \'' + pool.prioritySubprojectAliases[0] + '\'' : 's: '
+                  + pool.prioritySubprojectAliases.join(', ')),
+          tableMessage: 'PRIORITY (' + (pool.prioritySubprojectAliases.length == 1 ? pool.prioritySubprojectAliases[0] : 'MULTIPLE') + ')',
+          tileMessage: 'PRIORITY ('
+              + (pool.prioritySubprojectAliases.length == 1 ? pool.prioritySubprojectAliases[0] : pool.prioritySubprojectAliases.join(', '))
+              + ')',
+          level: 'info'
+        }, {
+          include: pool.duplicateIndices,
+          headerMessage: 'This pool contains duplicate indices!',
+          tableMessage: '(DUPLICATE INDICES)',
+          tileMessage: 'DUPLICATE INDICES'
+        }, {
+          include: pool.nearDuplicateIndices && !pool.duplicateIndices,
+          headerMessage: 'This pool contains near-duplicate indices!',
+          tableMessage: '(NEAR-DUPLICATE INDICES)',
+          tileMessage: 'NEAR-DUPLICATE INDICES'
+        }, {
+          include: pool.hasEmptySequence,
+          headerMessage: 'This pool contains at least one library with no index!',
+          tableMessage: '(MISSING INDEX)',
+          tileMessage: 'MISSING INDEX'
+        }, {
+          include: pool.hasLowQualityLibraries,
+          headerMessage: 'This pool contains at least one low quality library!',
+          tableMessage: '(LOW QUALITY LIBRARIES)',
+          tileMessage: 'LOW QUALITY LIBRARIES'
+        }, {
+          include: revokedDilutions.length > 0,
+          headerMessage: "Donor has revoked consent for " + revokedDilutions.toString(),
+          tableMessage: '(CONSENT REVOKED)',
```

Comment:
For some reason, I'm not able to get this warning to show up on the list pools table...

Reply:
The dilutions weren't being included in the PoolDto. This used to be a performance optimization, but the dilutions all get loaded now either way since we're checking for duplicate indices. I'll include them